### ISSUE TITLE
Support Speculative Execution

### ISSUE THREAD

**Issue by NikhilCollooru** [prestodb/presto#16493]
For a given query, some of the leaf level nodes might be slow in scanning and returning the data. This might be because
1) node is slow in communicating with underlying storage OR
2) node itself is slow due to faulty hardware etc. 
In such cases, instead of waiting for the slow node to finish the split, we can schedule the same work (split) to another node and pick the results from the node that finishes first among the two. This way we can reduce the long tail query latencies.

First step is identifying a retriable unit of work. In case of speculative execution for leaf level tasks, its the split. Split is the smallest unit of work that can be retried. 

Basically, goal of leaf level speculative execution is to improve the query latency by identifying the slow splits, scheduling copy of those slow splits on another worker and then picking the results of the node that finishes first among the two.
